src/DocumentFeedMigrator.cs

```diff
@@ -67,10 +70,24 @@ public DocumentFeedMigrator(Container containerToStoreDocuments)
                     log.LogMetric($"Failed items with StatusCode {failedMetric.Key}", failedMetric.Value);
                     totalFailedItems += failedMetric.Value;
                 }
-
                 log.LogMetric("Documents migrated", documents.Count - totalFailedItems);
                 log.LogMetric("Migration Time in ms", stopwatch.ElapsedMilliseconds);
             }
+
+        }
+
+        public Document MapPartitionKey(Document document, Document doc)
+        {
+            dynamic json = JsonConvert.DeserializeObject(doc.ToString());
+            StringBuilder syntheticKey = new StringBuilder();
+            string[] sourceAttributeArray = sourcePartitionKeyMapping.Split(',');
+            foreach (string attribute in sourceAttributeArray) {
+                syntheticKey.Append(json[attribute]);
+            }
+            json[targetPartitionKeyAttribute] = syntheticKey.ToString();
```

Comment:
What about just using document.GetPropertyValue and document.SetPropertyValue, that removes all the dynamic JSON parsing.